x-pack/legacy/plugins/apm/public/utils/flattenObject.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { compact, isObject } from 'lodash';
+
+export type FlattenItems = Array<{ key: string; value: unknown }>;
```

Comment:
It's mostly better to define it as singularL

```suggestion
export type KeyValuePair = { key: string; value: unknown };
```

And then use it as `KeyValuePair[]`